controllers/toolchaincluster/healthchecker_test.go

```diff
@@ -39,139 +33,50 @@ func TestClusterHealthChecks(t *testing.T) {
-			clusterconditions: []toolchainv1alpha1.Condition{offline()},
-			status:            withStatus(healthy()),
-		},
-		//if the connection cannot be established at beginning, then it should be offline
-		"OfflineConnectionNotEstablished": {
-			tctype:            "failing",
-			apiendpoint:       "http://failing.com",
-			clusterconditions: []toolchainv1alpha1.Condition{offline()},
-			status:            toolchainv1alpha1.ToolchainClusterStatus{},
-		},
-		//if no zones nor region is retrieved, then keep the current
-		"NoZoneKeepCurrent": {
-			tctype:            "stable",
-			apiendpoint:       "http://cluster.com",
-			clusterconditions: []toolchainv1alpha1.Condition{healthy()},
-			status:            withStatus(offline()),
+		"ErrorWhileDoingHealth": {
+			tctype:      "Notfound",
+			apiendpoint: "http://not-found.com",
+			healthcheck: false,
 		},
 	}
 	for k, tc := range tests {
 		t.Run(k, func(t *testing.T) {
-			tctype, sec := newToolchainCluster(tc.tctype, tcNs, tc.apiendpoint, tc.status)
+			//given
+			tctype, sec := newToolchainCluster(tc.tctype, tcNs, tc.apiendpoint, toolchainv1alpha1.ToolchainClusterStatus{})
 			cl := test.NewFakeClient(t, tctype, sec)
 			reset := setupCachedClusters(t, cl, tctype)
 			defer reset()
 			cachedtc, found := cluster.GetCachedToolchainCluster(tctype.Name)
 			require.True(t, found)
 			cacheclient, err := kubeclientset.NewForConfig(cachedtc.RestConfig)
 			require.NoError(t, err)
-			healthChecker := &HealthChecker{
-				localClusterClient:     cl,
-				remoteClusterClient:    cachedtc.Client,
-				remoteClusterClientset: cacheclient,
-				logger:                 logger,
-			}
-			// when
-			err = healthChecker.updateIndividualClusterStatus(context.TODO(), tctype)
 
-			//then
-			require.NoError(t, err)
-			assertClusterStatus(t, cl, tc.tctype, tc.clusterconditions...)
-		})
-	}
-}
+			//when
+			healthcheck, errh := getClusterHealthStatus(context.TODO(), cacheclient)
 
-func withStatus(conditions ...toolchainv1alpha1.Condition) toolchainv1alpha1.ToolchainClusterStatus {
-	return toolchainv1alpha1.ToolchainClusterStatus{
-		Conditions: conditions,
-	}
-}
-func assertClusterStatus(t *testing.T, cl client.Client, clusterName string, clusterConds ...toolchainv1alpha1.Condition) {
-	tc := &toolchainv1alpha1.ToolchainCluster{}
-	err := cl.Get(context.TODO(), test.NamespacedName("test-namespace", clusterName), tc)
-	require.NoError(t, err)
-	assert.Len(t, tc.Status.Conditions, len(clusterConds))
-ExpConditions:
-	for _, expCond := range clusterConds {
-		for _, cond := range tc.Status.Conditions {
-			if expCond.Type == cond.Type {
-				assert.Equal(t, expCond.Status, cond.Status)
-				assert.Equal(t, expCond.Reason, cond.Reason)
-				assert.Equal(t, expCond.Message, cond.Message)
-				continue ExpConditions
+			//then
+			require.Equal(t, tc.healthcheck, healthcheck)
+			if tc.tctype == "Notfound" {
+				require.Error(t, errh)
+			} else {
+				require.NoError(t, errh)
 			}
```

Comment:
you could use the `tc.errh` (because you don't use it at all), set it as:
```
		"ErrorWhileDoingHealth": {
			tctype:      "Notfound",
			apiendpoint: "http://not-found.com",
			healthcheck: false,
			errh:        fmt.Errorf("the server could not find the requested resource"),
		},
```

and then run here

```suggestion
			if tc.errh != nil {
				require.EqualError(t, errh, tc.errh.Error())
			} else {
				require.NoError(t, errh)
			}
```